cli/command/stack/kubernetes/services.go

```diff
@@ -2,43 +2,130 @@ package kubernetes
 
 import (
 	"fmt"
+	"sort"
+	"strings"
 
 	"github.com/docker/cli/cli/command/formatter"
 	"github.com/docker/cli/cli/command/stack/options"
 	"github.com/docker/cli/kubernetes/labels"
+	"github.com/docker/docker/api/types/filters"
+	appsv1beta2 "k8s.io/api/apps/v1beta2"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// RunServices is the kubernetes implementation of docker stack services
-func RunServices(dockerCli *KubeCli, opts options.Services) error {
-	// Initialize clients
+var supportedServicesFilters = map[string]bool{
+	"mode":  true,
+	"name":  true,
+	"label": true,
+}
+
+func generateValuedLabelsSelector(valuedLabels map[string][]string) []string {
+	var result []string
+	for k, v := range valuedLabels {
+		if len(v) == 1 {
+			result = append(result, fmt.Sprintf("%s=%s", k, v[0]))
+		} else {
+			sort.Strings(v)
+			result = append(result, fmt.Sprintf("%s in (%s)", k, strings.Join(v, ",")))
+		}
+	}
+	return result
+}
+
+func parseLabelFilters(rawFilters []string) ([]string, map[string][]string) {
```

Comment:
Is there any code in the Kubernetes CLI we could reuse here?

Reply:
What do you mean by Kubernetes CLI? kubectl?

Reply:
Yeah, I'm wondering if there are any public functions in the `kubernetes/kubernetes` repo we could use here.

Reply:
I think this code transforms label filters (with swarm syntax) to their Kubernetes counterpart. I don't think there is any function in `kubernetes/kubernetes` for that, as the `kubectl` user will write its filters directly in Kubernetes syntax, which are resolved server side IIRC so `kubectl` doesn't have to parse/interpret it.